api/src/main/java/io/strimzi/api/kafka/model/status/ListenerStatus.java

```diff
@@ -28,19 +28,20 @@
         builderPackage = Constants.FABRIC8_KUBERNETES_API
 )
 @JsonInclude(JsonInclude.Include.NON_NULL)
-@JsonPropertyOrder({ "type", "addresses", "bootstrapServers", "certificates" })
+@JsonPropertyOrder({ "type", "name", "addresses", "bootstrapServers", "certificates" })
 @EqualsAndHashCode
 public class ListenerStatus implements UnknownPropertyPreserving, Serializable {
     private static final long serialVersionUID = 1L;
 
     private String type;
+    private String name;
     private List<ListenerAddress> addresses;
     private String bootstrapServers;
     private List<String> certificates;
     private Map<String, Object> additionalProperties;
 
-    @Description("The type of the listener. " +
-            "Can be one of the following three types: `plain`, `tls`, and `external`.")
+    @Description("The `type` field in `ListenerStatus` have been deprecated and will be removed when updating to CRD v1" +
+            "The type of the listener can be one of the following three types: `plain`, `tls`, and `external`.")
```

Comment:
You should use the `@DeprecatedProperty` annotation. 

```suggestion
    @Description("The name of the listener.")
    @DeprecatedProperty(description="The `type` property is deprecated and has been replaced with the `name` property.", removalVersion="v1", movedToPath="name")
```

Reply:
@tombentley Actually I tried using `@DeprecatedProperty earlier but the issue with it is that have to use @Deprecated annotation too then. And if I use it the test eventually starts failing since an extra condition gets added --> ```Condition(status=True, reason=DeprecatedFields, message=In API version null the type property at path status.listeners.type has been deprecated. This field has been deprecated and will be removed when upgrading to CRD
 v1., type=Warning, lastTransitionTime=2021-11-08T06:43:24.066635110Z, additionalProperties={})```

Reply:
Why can't you change the test to expect that warning?

Reply:
I had a discussion with Jakub regarding this and he told that we might not want it